gonat_linux.go

```diff
@@ -246,26 +285,28 @@ func (s *server) assignPort(downFT FiveTuple) (upFT FiveTuple, err error) {
 }
 
 func (s *server) reapIdleConns() {
-	var connsToClose []*conn
 	for _, c := range s.connsByDownFT {
 		if c.timeSinceLastActive() > s.opts.IdleTimeout {
-			connsToClose = append(connsToClose, c)
-			s.deleteConn(c)
+			s.deleteAndCloseConn(c)
 		}
 	}
-	if len(connsToClose) > 0 {
-		// close conns on a goroutine to avoid tying up main dispatch loop
-		ops.Go(func() {
-			for _, c := range connsToClose {
-				c.Close()
-			}
-		})
+}
+
+func (s *server) requestCloseConn(c *conn) {
+	s.closingConns <- c
+}
+
+func (s *server) deleteAndCloseConn(c *conn) {
+	if c.doClose() {
+		delete(s.connsByDownFT, c.downFT)
+		delete(s.connsByUpFT, c.upFT)
```

Comment:
Nit: deleting a non-existent key from a map is a no-op, so the `if` check shouldn't be necessary (and therefore `doClose` does not need a return value).

Reply:
Haha, it was left over from when I was also incrementing closingConns! Will fix.

Reply:
Oh, actually I remembered why I have this here. The map is keyed to the five tuple. Once we've closed a conn with that five tuple, it's theoretically possible for us to create a new Conn with that five tuple, which will get stored in that map. If something else tries to subsequently close the old conn again, we'll end up deleting the live conn from the map. I'll add a comment about that.